File: python/字符串/266. 回文排列.py
class Solution:
    def canPermutePalindrome(self, s: str) -> bool:
         l = len(s)
         from collections import defaultdict
         d = defaultdict(int)
         for i in s:
             d[i]+=1
         if l % 2 == 0:
             for k,v in d.items():
                 if v % 2 != 0:
                     return False
             return True
         else:
             count = 0
             for k,v in d.items():
                 if v % 2 != 0:
                     count+=1
                 if count > 1:
                     return False
             return True

File: go/string/266. 回文排列.go
package string

func canPermutePalindrome(s string) bool {
	d := make(map[byte]int)
	for i := 0; i < len(s); i++ {
		d[s[i]]++
	}
	l := len(s)
	if l%2 == 0 {
		for _, v := range d {
			if v%2 != 0 {
				return false
			}
		}
		return true
	} else {
		count := 0
		for _, v := range d {
			if v%2 != 0 {
				count++
			}
			if count > 1 {
				return false
			}
		}
		return true
	}
}
